File: common/src/main/java/com/pluxity/label3d/Label3DResponse.java
package com.pluxity.label3d;

import com.pluxity.feature.entity.Spatial;

public record Label3DResponse(
        String id,
        String displayText,
        String floorId,
        Spatial position,
        Spatial rotation,
        Spatial scale) {
    public static Label3DResponse from(Label3D label3D) {
        return new Label3DResponse(
                label3D.getId(),
                label3D.getDisplayText(),
                label3D.getFeature() != null ? label3D.getFeature().getFloorId() : null,
                label3D.getFeature() != null ? label3D.getFeature().getPosition() : null,
                label3D.getFeature() != null ? label3D.getFeature().getRotation() : null,
                label3D.getFeature() != null ? label3D.getFeature().getScale() : null);
    }
}

File: common/src/main/kotlin/com/pluxity/label3d/Label3DResponse.kt
package com.pluxity.label3d

import com.pluxity.feature.entity.Spatial

data class Label3DResponse(
    val id: String,
    val displayText: String,
    val floorId: String?,
    val position: Spatial?,
    val rotation: Spatial?,
    val scale: Spatial?,
)

fun Label3D.toLabel3DResponse(): Label3DResponse =
    Label3DResponse(
        id = this.requiredId(),
        displayText = requireNotNull(this.displayText) { "displayText is null (not ready)" },
        floorId = this.feature.floorId,
        position = this.feature.position,
        rotation = this.feature.rotation,
        scale = this.feature.scale,
    )
